manifest.go

```diff
@@ -67,8 +67,13 @@ func validateManifest(s string) ([]error, error) {
 					for key, value := range v.(map[string]interface{}) {
 						// Check if the key is valid
 						switch key {
-						case "name", "branch", "revision", "version", "source":
+						case "name", "branch", "version", "source":
 							// valid key
+						case "revision":
+							// Check if sha1 hash is abbreviated
+							if valueStr, ok := value.(string); ok && len(valueStr) < 40 {
```

Comment:
Pretty sure it's a revision-id, see https://github.com/golang/dep/blob/master/cmd/dep/ensure_test.go#L24 for what we are validating against in our tests. I think that's a revision-id, yeah?